tests/scp/perform_scp.py

```diff
@@ -0,0 +1,35 @@
+import pexpect
+import sys
+
+is_source_ip = sys.argv[1] 
+ip = sys.argv[2]
+source_path = sys.argv[3]
+dest_path = sys.argv[4]
+user = sys.argv[5]
+
+command = ""
+if is_source_ip == "y":
+    command = "scp {}@{}:{} {}".format(user, ip, source_path, dest_path)
+else:
+    command = "scp {} {}@{}:{}".format(source_path, user, ip, dest_path)
+
+
+try:
+    #child = pexpect.spawn("scp root@10.250.0.102:/root/test_file.bin /home/admin")
+    child = pexpect.spawn(command)
+    prompt = child.expect(["Are you sure you want to continue connecting (yes/no)?", "password:"])
+
+    if prompt == 0:
+        child.sendline("yes")
+        child.expect("password")
+
+    child.sendline("root")
```

Comment:
might be we can read the password from creds fixture? such as
```
dutpass = creds['sonicadmin_password']
```

Reply:
I tried looking for this by using creds - however, could not find root or ptf_password. I'm not sure how to include in the creds variable - I'm assuming I have to somehow add in a yml and make sure ansible adds it into the creds variable. Is there a reference I can go through to add it in?

Reply:
It should have been included, that is how we access ptf host in test. But because you are executing this test from DUT perspective, so you don't have access to that group var.

Reply:
I agree with @Blueve that password should not be hard coded here. Since this script is to be executed on the DUT, it does not have direct access to the `creds` fixture. It can only get PTF host username and password through script arguments, like the other arguments ip, user, path, etc.
In script `test_scp_copy.py` you should have access to the `creds` fixture. The PTF host username and password are defined in `ansible/group_vars/ptf_host/secrets.yml`. You can try `creds[‘ptf_host_user’]` and `creds[‘ptf_host_pass’]`. For this to work, you need to have your PTF hosts defined under group `ptf_host` in your inventory file.

Reply:
Thanks - there was a bit of miscommunication but the offline discussion definitely helped. I copied the credentials to group_vars/all/creds.yml and was able to access the password from the test through the creds variable after that. Sent the password to the script as an additional variable to avoid the hard-copy. Resolving. Please feel free to re-open this conversation if required.